tracer/src/Datadog.Trace/Agent/StatsAggregator.cs

```diff
@@ -159,15 +167,17 @@ private static double ConvertTimestamp(long ns)
             return ns << shift;
         }
 
-        private void AddToBuffer(Span span)
+        private bool AddToBuffer(Span span)
         {
             if ((!span.IsTopLevel && span.GetMetric(Tags.Measured) != 1.0) || span.GetMetric(Tags.PartialSnapshot) > 0)
             {
-                return;
+                return false;
             }
 
             var key = BuildKey(span);
 
+            var isNewKey = _keys.Add(key);
```

Comment:
From the [trace ingestion mechanism doc](https://datadoghq.atlassian.net/wiki/spaces/APM/pages/2564915820/Trace+Ingestion+Mechanisms#RareSampler), it seems we need to add one tag in that case:

> _dd.rare: Sampling rate of the RareSampler